### ISSUE TITLE
Please support the files saved in China (currently only ['en_US.utf8', 'de_DE', 'fr_FR', 'it_IT')

### ISSUE THREAD

**Issue by PullGuy** [mne-tools/mne-nirs#543]
**READ THIS FIRST:** If you are having trouble getting MNE-Python to work with
your own data, you should ask for help on one of our other channels:

- [email list](https://mail.nmr.mgh.harvard.edu/mailman/listinfo/mne_analysis)
- [Gitter (chat)](https://gitter.im/mne-tools/mne-python)

Our GitHub issue tracker is only used to report bugs and suggest improvements
to MNE-Python. For any other questions, please use the email list or Gitter.
Usage questions that are posted as GitHub issues are usually closed without
being answered. See
[the FAQ entry on filing bug reports](https://mne.tools/dev/overview/faq.html#i-think-i-found-a-bug-what-do-i-do)
for more guidance. If you're pretty sure your problem is a bug in MNE-Python,
please **delete this section** and fill in the headings below, replacing the
placeholder text with your own responses. Surround any code samples with triple
backticks above and below the code block (see
[the GitHub guide to markdown](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown)
for help with issue formatting). Alternatively, you can put your MWE in a
[public gist](https://gist.github.com) and link to it in this issue.

#### Describe the bug

I'm in China, and I used a NIRSport device to collect fNIRs data, where the date information is in a Chinese language format.
When I read data like this:
`data = mne.io.read_raw_nirx(r'C:UsersDuckDocumentsGithubfNIRs-Data-ProcessSubject1')` 
Python returns an error message:

RuntimeWarning: Extraction of measurement date from NIRX file failed. This can be caused by files saved in certain locales (currently only ['en_US.utf8', 'de_DE', 'fr_FR', 'it_IT'] supported). Please report this as a github issue. The date is being set to January 1st, 2000, instead of '"2021Äê7ÔÂ3ÈÕ""16:33:23.262"'.

I looked at the *.hdr file where the date information is：`Date="2021年7月3日"`

It also gives the following error:

`ValueError                                Traceback (most recent call last)
Cell In[47], [line 1](vscode-notebook-cell:?execution_count=47&line=1)
----> [1](vscode-notebook-cell:?execution_count=47&line=1) data = read_raw_nirx(r"c:\Users\DUCK\Documents\Github\fNIRs-Data-Process\Subject1").load_data()

File <decorator-gen-206>:12, in load_data(self, verbose)

File [c:\Users\DUCK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:592](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:592), in BaseRaw.load_data(self, verbose)
    [573](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:573) """Load raw data.
    [574](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:574) 
    [575](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:575) Parameters
   (...)
    [589](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:589) .. versionadded:: 0.10.0
    [590](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:590) """
    [591](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:591) if not self.preload:
--> [592](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:592)     self._preload_data(True)
    [593](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:593) return self

File [c:\Users\DUCK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:604](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:604), in BaseRaw._preload_data(self, preload)
    [599](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:599)     data_buffer = None
    [600](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:600) logger.info(
    [601](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:601)     "Reading %d ... %d  =  %9.3f ... %9.3f secs..."
    [602](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:602)     % (0, len(self.times) - 1, 0.0, self.times[-1])
    [603](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:603) )
--> [604](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:604) self._data = self._read_segment(data_buffer=data_buffer)
...
--> [566](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:566) x.shape = (stop - start, -1)
    [567](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:567) x = x[:, cols]
    [568](file:///C:/Users/DUCK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:568) return x

ValueError: cannot reshape array of size 1 into shape (18885,newaxis)`

#### Steps to reproduce

[MWE]: https://en.wikipedia.org/wiki/Minimal_Working_Example
[built-in datasets]: https://mne.tools/dev/overview/datasets_index.html
[sample]: https://mne.tools/dev/overview/datasets_index.html#sample

#### Expected results

I want it to read the data correctly and do something else

#### Actual results
*Replace this text with the actual output, traceback, screenshot, or other
description of the results.*

**Comment by drammock** [comment]
Through a bit of trial and error I was able to figure out what encoding is used in your file:

```pycon
In [1]: foo = "2021Äê7ÔÂ3ÈÕ""16:33:23.262"  # what you pasted in the GitHub issue
In [2]: foo.encode("windows-1252").decode("GB18030")
Out[2]: '2021年7月3日16:33:23.262'
```

I think probably allowing the user to specify what encoding is used will be part of the solution, and the other part is making MNE smarter about parsing a measurement date --- i.e., making it know what 年 (year), 月 (month), and 日 (day) mean.

It would be helpful to have the *full* traceback, you've omitted some steps here:
```
--> 604 self._data = self._read_segment(data_buffer=data_buffer)
...
--> 566 x.shape = (stop - start, -1)
```

Even more helpful would be to have a link to the file, if you're able to share it.

**Comment by CaoKang-David** [comment]
OK, the data file is attached here.  [Subject1.zip](https://github.com/mne-tools/mne-nirs/files/14782184/Subject1.zip)

In my opinion, a better solution would be to read the *.nirs file directly, as in MATLAB, and the extraction of date information should not affect other data processing that followed.

and,  why does this error happen? `ValueError: cannot reshape array of size 1 into shape (18885,newaxis)`，Is it because of a date information processing error?

**Comment by CaoKang-David** [comment]
The Full TraceBack are here:
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[27], [line 1](vscode-notebook-cell:?execution_count=27&line=1)
----> [1](vscode-notebook-cell:?execution_count=27&line=1) data.load_data()

File <decorator-gen-206>:12, in load_data(self, verbose)

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:592](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:592), in BaseRaw.load_data(self, verbose)
    [573](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:573) """Load raw data.
    [574](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:574) 
    [575](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:575) Parameters
   (...)
    [589](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:589) .. versionadded:: 0.10.0
    [590](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:590) """
    [591](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:591) if not self.preload:
--> [592](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:592)     self._preload_data(True)
    [593](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:593) return self

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:604](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:604), in BaseRaw._preload_data(self, preload)
    [599](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:599)     data_buffer = None
    [600](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:600) logger.info(
    [601](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:601)     "Reading %d ... %d  =  %9.3f ... %9.3f secs..."
    [602](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:602)     % (0, len(self.times) - 1, 0.0, self.times[-1])
    [603](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:603) )
--> [604](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:604) self._data = self._read_segment(data_buffer=data_buffer)
    [605](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:605) assert len(self._data) == self.info["nchan"]
    [606](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:606) self.preload = True

File <decorator-gen-205>:12, in _read_segment(self, start, stop, sel, data_buffer, verbose)

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:480](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:480), in BaseRaw._read_segment(***failed resolving arguments***)
    [478](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:478)     # reindex back to original file
    [479](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:479)     orig_idx = _convert_slice(self._read_picks[fi][need_idx])
--> [480](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:480)     _ReadSegmentFileProtector(self)._read_segment_file(
    [481](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:481)         data[:, this_sl],
    [482](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:482)         orig_idx,
    [483](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:483)         fi,
    [484](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:484)         int(start_file),
    [485](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:485)         int(stop_file),
    [486](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:486)         cals,
    [487](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:487)         mult,
    [488](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:488)     )
    [489](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:489)     offset += n_read
    [490](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:490) return data

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\base.py:2623](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:2623), in _ReadSegmentFileProtector._read_segment_file(self, data, idx, fi, start, stop, cals, mult)
   [2622](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:2622) def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
-> [2623](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:2623)     return self.__raw.__class__._read_segment_file(
   [2624](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:2624)         self, data, idx, fi, start, stop, cals, mult
   [2625](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/base.py:2625)     )

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\nirx\nirx.py:534](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:534), in RawNIRX._read_segment_file(self, data, idx, fi, start, stop, cals, mult)
    [532](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:532) wls = list()
    [533](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:533) for key in ("wl1", "wl2"):
--> [534](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:534)     d = _read_csv_rows_cols(
    [535](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:535)         self._raw_extras[fi]["files"][key],
    [536](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:536)         start,
    [537](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:537)         stop,
    [538](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:538)         sd_index,
    [539](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:539)         self._raw_extras[fi]["bounds"][key],
    [540](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:540)     ).T
    [541](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:541)     nan_mask = self._raw_extras[fi]["nan_mask"].get(key, None)
    [542](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:542)     if nan_mask is not None:

File [c:\Users\CK\AppData\Local\Programs\Python\Python311\Lib\site-packages\mne\io\nirx\nirx.py:566](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:566), in _read_csv_rows_cols(fname, start, stop, cols, bounds, sep, replace)
    [564](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:564)         data = replace(data)
    [565](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:565)     x = np.fromstring(data, float, sep=sep)
--> [566](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:566) x.shape = (stop - start, -1)
    [567](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:567) x = x[:, cols]
    [568](file:///C:/Users/CK/AppData/Local/Programs/Python/Python311/Lib/site-packages/mne/io/nirx/nirx.py:568) return x

ValueError: cannot reshape array of size 1 into shape (18885,newaxis)

**Comment by drammock** [comment]
sorry for the long wait on this one!  With MNE-Python main (and with the 1.8 release coming out next month), it will be possible to pass `read_raw_nirx(..., encoding="GB18030")` to read these files successfully.

Regarding your second question about `ValueError: cannot reshape array of size 1 into shape (18885,newaxis)` --- I cannot replicate that error happening with the zipped version of the file you shared.  If you're still experiencing that problem on MNE-Python current main (or on 1.8 after it comes out), please open a new issue about it.
